### DISCUSSION TITLE
Possible bug with API key(s)

### DISCUSSION THREAD

**Discussion by elerik** [OP-TED/eForms-SDK#299] [Q&A] [answered]
Hi devs 
May be Iam wrong and i missed something somewhere or i made some funny problem in between keyboard and chair. Anyway its work in progress and still chill so lets get in to it.

I send request at "https://enotices2.preview.ted.europa.eu/esenders/api/v2/notice/notices/search-esentool" with active API key. Everything is ok. I get all what i want in response. 
If i use this API key for request at "https://cvs.ted.europa.eu/v1/notices/validation" i have invalid x-api-key message response

So i use my previous API key which should be invalid/expired. But it works and i get my message with errors in sended xml which is right response. But wrong implementation because this API key should not be active. 

What happend ? 
I revoked my API key at developer.ted.europa.eu. From that moment I have this awesome mix of api keys so i have to store both API keys in my app for playing with ted servers. 
Active one for enotices2.preview.ted.europa.eu and expired one for cvs.ted.europa.eu. Interesting isnt it ? 😅

**Comment by bertrand-lorentz** [comment, marked as answer]
Hi,

I think your issue is because you are mixing up the 2 environments that are available to you:
* Production
* Preview, to be used for testing your developments

The API keys for each environment are distinct, and have to be obtained from the Developer Portal in the corresponding environment.
See https://docs.ted.europa.eu/home/eforms/preview/index.html

All apps and APIs in the "preview" environment have "preview" in their domain name.
So it seems you're using eNotices2 in "preview", but CVS in "production". CVS in "preview" will accept the same key as eNotices2.

**Reply by elerik** [reply]
Awesome. Confirm. I didnt noticed that cvs has also preview version.  👍 thanks
